design/use-statement.md

```diff
@@ -0,0 +1,162 @@
+If this behavior isn't desired, it can be overridden with:
+```yaml
+- use: someTool someVersion
+  proxy: false
+```
+
+If a tool needs credentials, those can be added like this:
+```yaml
+resources:
+  feeds:
+  - feed: myFeed
+    auth: {auth_information}
+
+steps:
+- use: someTool someVersion
+  auth: myFeed
+```
+
+`feed` is not yet a resource type, but it's coming.
+
+No additional keywords are reserved.
+Any additional keywords are passed through as `inputs` to the underlying task.
+
+## Schema
+
+```yaml
+- use: string       # required tool name and optional version
+  version: string   # optional version (specifying version twice is an error)
+  proxy: boolean    # whether to install proxy information; defaults to true
+  auth: string      # optional name of a resource to derive auth information from
+  {string: any}     # additional arguments to pass as task inputs
+```
+
+## Example
+
+```yaml
+strategy:
+  matrix:
+    PY35-NODE8:
+      py_version: '3.5'
+      node_version: '8'
+    PY35-NODE10:
+      py_version: '3.5'
+      node_version: '10'
+    PY27-NODE8:
+      py_version: '2.7'
+      node_version: '8'
+    PY27-NODE10:
+      py_version: '2.7'
+      node_version: '10'
+
+steps:
+- use: python $(py_version)
+- use: node $(node_version)
+- script: npm install ...
+- script: python ...
+```
+
+*Note, as there's been confusion:* nothing changes in the `matrix` section.
+It's purely about the `steps`.
+
+### Task changes
+
+The in-box tool installers and Use*Tools will evolve.
+
+Task requirements:
+- task.json includes a field indicating what ecosystem the task provides.
+Something like `{ 'provides': string }`.
+- Accept `version` as an input.
+By convention, this should be a SemVer version spec.
+It's up to the task to process these (with help from the task lib).
+If an ecosystem has additional, non-SemVer "versions", the task may accept those as well.
+To support container jobs, `version` is optional.
+It's strongly recommended for VM jobs.
+- Document the range of acceptable version numbers.
+Where it's "install from the internet", say so.
+Where it's "from a list in the hosted tools cache", that should be clear in the docs.
+Also, we'll make sure all the tooling includes this information - for instance, Intellisense.
+  - Future work: some way to acquire the list of available versions at runtime.
+- Some tools have multiple architectures available.
+By convention, an `architecture` field should use the agent architecture types: `x64`, `x86`, and `arm`.
+Others can be added, and of course only the relevant ones should be accepted.
+
+Additional requirements for in-box tasks:
+- By convention, in-box tasks will use a consistent name scheme: `Use<Name>`.
+This won't be required for third-party tasks.
+- Another common parameter is `architecture`.
+- Do a simplification pass on each task to make sure each input is necessary, clear, and orthogonal.
+
+## Appendix: List of existing tool and installer tasks
```

Comment:
what happens when i rev the major-version for my task?

Reply:
Using this syntax opts you into binding the latest major version. Inputs which aren't accepted by the task should add a build warning (so that task authors have a mechanism to remove inputs that won't automatically break the world).